resources/assets/components/Signup/index.js

```diff
@@ -296,24 +299,26 @@ class Signup extends React.Component {
     }
   }
 
-  // Submit a new reportback on behalf of the user.
-  submitReportback(reportback) {
-    // Fields to send to /posts
+  // @TODO: make this work for any type of post?
+  // Submit a new photo post on behalf of the user.
+  submitPost(post) {
+    // Fields to send to /v3/posts
     const fields = {
-      northstar_id: reportback.northstarId,
-      campaign_id: reportback.campaignId,
-      campaign_run_id: reportback.campaignRunId,
-      quantity: reportback.impact,
-      why_participated: reportback.whyParticipated,
-      caption: reportback.caption,
-      source: reportback.source,
-      status: reportback.status,
-      file: reportback.media.dataURL,
-      remote_addr: '',
+      northstar_id: post.northstarId,
+      campaign_id: post.campaignId,
+      campaign_run_id: post.campaignRunId,
+      quantity: post.impact,
+      why_participated: post.whyParticipated,
+      text: post.text,
+      source: post.source,
+      status: post.status,
+      file: post.media.dataURL,
```

Comment:
We want to upload the actual file here, not a Data URL string.

I believe that's `post.media.file` (set by the [MediaUploader](https://github.com/DoSomething/rogue/blob/master/resources/assets/components/MediaUploader/index.js)).

Reply:
Let's also make sure we're on the latest Gateway by running an `npm update`!

Reply:
ah! I changed that but still getting this: 
![screen shot 2018-04-06 at 10 48 57 am](https://user-images.githubusercontent.com/9019452/38427815-3e213dda-3988-11e8-9545-af7946614a14.png)

Reply:
Can you see what's in the details of the 422 response (click the "arrow" on the left of the `GATEWAY: Uprocessable Entity` error? Is it still complaining about the `file` field?

I'm also curious what the request payload looks like - you should be able to find the post request in Chrome's web inspector's "Network" tab, then click on it and scroll to the bottom of the "headers" tab to see what's in "Request Payload".

Reply:
ah so interesting I didn't know you could do that! 

Here's what the 422 response says when I click the "arrow":
![screen shot 2018-04-06 at 11 03 01 am](https://user-images.githubusercontent.com/9019452/38428771-9248c8ae-398a-11e8-897a-95dc1e025ae0.png)

Digging around I'm seeing some inconsistencies though. Here's the response that says I didn't pass these fields but they are definitely being passed...
![screen shot 2018-04-06 at 11 05 04 am](https://user-images.githubusercontent.com/9019452/38428797-a7a429fa-398a-11e8-94c2-2d4528b13ec7.png)

Then it says that I'm not sending `multipart/form-data` when we just updated so it should be...
![screen shot 2018-04-06 at 11 05 50 am](https://user-images.githubusercontent.com/9019452/38428834-c315e0d4-398a-11e8-8be6-9e15cab37c2b.png)

And lastly, it looks like the file is empty? 
![screen shot 2018-04-06 at 11 06 01 am](https://user-images.githubusercontent.com/9019452/38428857-cb41e140-398a-11e8-8c33-8473cd406762.png)

Reply:
looks like I am sending the blob in the request (`console.log(fields)` before sending the request): 
![screen shot 2018-04-06 at 11 10 41 am](https://user-images.githubusercontent.com/9019452/38428998-37029cbc-398b-11e8-83ef-c26bd21ddce4.png)